src/db.c

```diff
@@ -190,7 +190,14 @@ robj *lookupKeyWriteOrReply(client *c, robj *key, robj *reply) {
     return o;
 }
 
-/* Add the key to the DB. It's up to the caller to increment the reference
+/* Add the key to the DB.
+ *
+ * The kvstore handles `key` based on `dictType` during initialization:
+ * - If `dictType.embedded-entry` is 1, it clones the `key`.
+ * - Otherwise, it assumes ownership of the `key`.
+ * In this case a copy of `key` is made in kvstore, the caller must ensure the `key` is properly freed.
```

Comment:
This function (`dbAddInternal`) operates on the dbDictType, so always copies it, right?